src/login/logind-session.c

```diff
@@ -170,23 +170,16 @@ void session_set_user(Session *s, User *u) {
         user_update_last_session_timer(u);
 }
 
-int session_set_leader(Session *s, pid_t pid) {
-        _cleanup_(pidref_done) PidRef pidref = PIDREF_NULL;
+int session_set_leader(Session *s, PidRef leader) {
         int r;
 
         assert(s);
+        assert_se(pidref_is_set(&leader));
```

Comment:
Why `assert_se`?

Reply:
You are right there is no reason for `assert_se()` here. Fixed.